src/hypothesis/internal/validation.py

```diff
@@ -127,15 +127,16 @@ def check_valid_interval(lower_bound, upper_bound, lower_name, upper_name):
 
 @check_function
 def check_valid_sizes(min_size, average_size, max_size):
+    if average_size is not None:
+        from hypothesis._settings import note_deprecation
+        note_deprecation(
+            'The average_size argument has been disabled, is deprecated, '
```

Comment:
It might be worth making this message slightly more explicit about what the user should do: e.g. "You should remove this argument as it no longer has any effect"